### PR TITLE
AVX-512 accelerated batch SHA512 implementation, related features and minor cleanups

### PR DESCRIPTION
**Pull request by kbowers-jump**
On a 2.3 GHz Icelake server core with turbo disabled (same one mentioned previously for AVX-512 accelerated ED25519), quick and dirty benchmarks saw a large batch asymptotic Ethernet equivalent bandwidth processing rate of:
```
impl   | small msg | large msg
-------+-----------+----------
ref    |  1.9 Gbps |  3.4 Gbps
avx    |  6.3 Gbps | 10.7 Gbps
avx512 |  9.0 Gbps | 15.6 Gbps
```
The small message size was 14 bytes (i.e. a maximum UDP payload of a minimal Ethernet frame holding a UDP/IP4/VLAN/Ethernet packet) and the large message size was 1472 bytes (i.e. a maximum UDP payload of a non-Jumbo 1500 byte MTU Ethernet frame holding a UDP/IP4/VLAN/Ethernet packet).

This is based on and extends the AVX-512 support added recently for faster ED25519 signature verification.

This implementation is a turn-the-crank widened version of the existing AVX implementation with some additional streamlining to take advantage of AVX-512 support for vector lane masks (this has been _obviously_ missing from the x86 vector ISAs for _decades_ and is one of the key microarchitectural features that makes GPUs effective ... in short, x86 cores and GPU "cores" / MPUs look increasingly similar at the low level ... just have to adjust the "warp" width from 128 bytes to 64 bytes and do the compilation of a CUDA-style implementation to vector intrinsics by hand).  That is, this is even more GPU-like than the existing AVX implementation under the hood.

Additionally, the SHA512 header was also tweaked to make it easier to switch implementations at compile time and/or in individual translation units without needing to change build targets.

Plan to do something similar for SHA256 shortly as accelerating ED25519, SHA512 and SHA256 frees up a lot of CPU resources in software only implementations for non-cryptographic usage.

Included here are lower level APIs necessary to implement this (e.g. 8-wide vector ulong support), support for the equivalent APIs for 8-wide vector longs to make that functionality increasingly complete and orthogonal, backports of faster versions of various AVX-512 operations to 32-byte and 16-byte wide vectors on build targets that support them, and updates to existing AVX-512 vector long code to use this functionality.  Along the way did some minor cleanups that ended up easier to include here than separate out into additional PRs (namely, stricter ED25519 testing and error handling, stricter brutality under gcc-13, FD_TLS -> FD_TL rename by popular demand to fix the quasi collision between "thread local storage" and "transport layer security").

TL;DR

AVX512 accelerated batch SHA512 is ~4.6x faster than the existing non-batched implementation and ~1.4x faster than the existing AVX accelerated implementation.  YMMV.

### CODE REVIEW

File: src/util/tpool/fd_tpool.cxx [hunk 1]
```diff
@@ -9,6 +9,10 @@ struct fd_tpool_private_worker_cfg {
 
 typedef struct fd_tpool_private_worker_cfg fd_tpool_private_worker_cfg_t;
 
+/* This is not not static to allow tile 0 to attach to this if desired. */
```
[the remark this answers is not in the document]
Comment (kbowers-jump) [reply src/util/tpool/fd_tpool.cxx:12]: It is a typo

[the remark this answers is not in the document]
Comment (jvarela-jump) [reply src/util/tpool/fd_tpool.cxx]: corrected

**Review by jvarela-jump — APPROVED** [review]
